PhysicsTools/Heppy/python/analyzers/gen/PDFWeightsAnalyzer.py

```diff
@@ -18,8 +18,9 @@ def __init__(self, cfg_ana, cfg_comp, looperName ):
     def declareHandles(self):
         super(PDFWeightsAnalyzer, self).declareHandles()
 
-        if self.doPDFWeights:
-            self.mchandles['pdfstuff'] = AutoHandle( 'generator', 'GenEventInfoProduct' )
+        #if self.doPDFWeights:
```

Comment:
we need the GenEventInfoProduct if we want to have x1, x2 etc. These are needed for the MEM calculation regardless if one is doing PDF systematics.
Otherwise, we can switch to have x1, x2, etc only in case we have the PDF weights. (is it the same as isMC?)